src/services/resource.js

```diff
@@ -389,12 +414,88 @@ module.exports = class resourceHelper {
 		)
 
 		const commentMapping = await comments.reduce((acc, item) => {
-			acc[item.resource_id] = parseInt(item.comment_count, 10) > 0 ? true : false
+			const programIds = programResourceObj?.programResourceMapping[item.resource_id]
+				? programResourceObj?.programResourceMapping[item.resource_id]
+				: []
+			if (programIds.length > 0) {
```

Comment:
@adithyadinesh0412  we can optimize 
if (programIds.length > 0) {
    programIds.forEach((programId) => {
        acc[programId] = parseInt(item.comment_count) > 0;
    });
} else {
    acc[item.resource_id] = parseInt(item.comment_count) > 0;
}